cpp/include/cudf/io/datasource.hpp

```diff
@@ -104,13 +104,21 @@ class datasource {
                                             size_t size   = 0);
 
   /**
-   * @brief Creates a source from a memory buffer.
+   * @brief Creates a source from a host memory buffer.
    *
    * @param[in] buffer Host buffer object
    * @return Constructed datasource object
    */
   static std::unique_ptr<datasource> create(host_buffer const& buffer);
 
+  /**
+   * @brief Creates a source from a device memory buffer.
+   *
+   * @param buffer Device buffer object
+   * @return Constructed datasource object
+   */
+  static std::unique_ptr<datasource> create(cudf::device_span<uint8_t> buffer);
```

Comment:
should be able to be const
```suggestion
  static std::unique_ptr<datasource> create(cudf::device_span<uint8_t const> buffer);
```